pennylane/measurements/mutual_info.py

```diff
@@ -76,4 +79,17 @@ def circuit_mutual(x):
 
     wires0 = qml.wires.Wires(wires0)
     wires1 = qml.wires.Wires(wires1)
-    return MeasurementProcess(MutualInfo, wires=[wires0, wires1], log_base=log_base)
+    return _MutualInfo(MutualInfo, wires=[wires0, wires1], log_base=log_base)
+
+
+class _MutualInfo(StateMeasurement):
+    """Measurement process that returns the mutual information."""
+
+    def process_state(self, state: np.ndarray, device_wires: Wires):
+        return qml.math.mutual_info(
+            state,
+            indices0=[self.wires[0]],
+            indices1=[self.wires[1]],
+            c_dtype=np.complex128,
```

Comment:
If a user was using an ML interface like torch or tf in their circuit, but measuring using `mutual_info`, would this force the numpy backend when returning the mutual information?

Reply:
Might be worth adding a test for

Reply:
Changed `c_dtype` to `"complex128"`. Added tests